src/pages/Search.jsx

```diff
@@ -1,7 +1,95 @@
+import React, { useState, useEffect } from 'react';
+import { useLocation } from 'react-router-dom';
+import { Image, Pagination } from '@nextui-org/react';
+import { Link } from 'react-router-dom';
+import api from '../configs/api';
+import Layout from '../layouts/Base';
+import toast from 'react-hot-toast';
+
 export default function SearchPage() {
+  // React-router-dom is used to access the URL query parameters.
+  const location = useLocation();
+  const query = new URLSearchParams(location.search);
```

Comment:
Here's a fun tip 😁: Have you tried the `useSearchParams` hook from `react-router-dom`? It could save us a line of code! 🎉 Just a bit of knowledge sharing - no pressure to change anything if you're happy with the current setup.

```javascript
const [searchParams, setSearchParams] = useSearchParams();

const keyword = searchParams.get('k');
const categorySlug = searchParams.get('c');
```